### PR TITLE
Azure storage

### PR DESCRIPTION
**Pull request by navjotk**
Use the following environment variables to use the Azure backend in production

STORAGE_PROVIDER=AZURE_BLOBS
STORAGE_KEY=icokpystorage
STORAGE_SECRET=eiO+9Vf6...gWjGXw==
STORAGE_CONTAINER=okpyfiles

### CODE REVIEW

**Comment by marrobi** [comment]
If the container doesn't exist waht is the behaviour? Do we need to precreate the container? This is a similar situation to the mySQL DB tables.

**Comment by marrobi** [comment]
Created new issue for precreation of container. Can we document in words the mapping of the above vars to Azure concepts as I got confused so others will.

**Comment by martinpeck** [comment]
The `test_simple` has been disabled within the Azure storage test suite because it's generating an issue when performing chunked downloaded of files. This looks like an issue in libcloud, and looks similar to another issue we spotted related to the upload of chunked files. More investigation needed, but if that proves to be the case we can turn this test back on when libcloud is fixed.

Having said that, this test is exercising code that is not used in ok.py. `file_download` within `files.py` only calls `storage.get_object_stream` for LOCAL storage providers. Hence, it's safe to skip the test for now.

File: server/storage.py [hunk 1]
```diff
@@ -148,7 +162,19 @@ def get_blob_url(self, obj_name, container_name=None, timeout=60):
...
+        """ 
+        Generates a signed URL compatible with Azure
+        Blob Storage
         """
-        return self.driver.download_object_as_stream(object)
+        # libcloud encodes the access key, so it needs decoding
+        secret_as_string = base64.b64encode(self.driver.secret).decode()
+
+        # azure expiry times need to be ISO8601 (no milliseconds, and with the timezone character)
+        expiry_datetime = expiry_datetime.replace(microsecond=0).isoformat() + 'Z'
```
Comment (martinpeck) [inline server/storage.py]: This code should be better. I'll raise an issue and see if there's a nicer way of doing this

Comment (c-w) [reply server/storage.py]: That code is fine IMO. DateTime formatting always ends up nasty.

**Pull request merged**